src/h3lib/lib/algos.c

```diff
@@ -395,27 +408,28 @@ H3Index h3NeighborRotations(H3Index origin, Direction dir, int *rotations) {
                 // base cell.
                 if (oldLeadingDigit == CENTER_DIGIT) {
                     // Undefined: the k direction is deleted from here
-                    return H3_NULL;
+                    return E_SUCCESS;
                 } else if (oldLeadingDigit == JK_AXES_DIGIT) {
                     // Rotate out of the deleted k subsequence
                     // We also need an additional change to the direction we're
                     // moving in
-                    out = _h3Rotate60ccw(out);
+                    working = _h3Rotate60ccw(working);
                     *rotations = *rotations + 1;
                 } else if (oldLeadingDigit == IK_AXES_DIGIT) {
                     // Rotate out of the deleted k subsequence
                     // We also need an additional change to the direction we're
                     // moving in
-                    out = _h3Rotate60cw(out);
+                    working = _h3Rotate60cw(working);
                     *rotations = *rotations + 5;
                 } else {
                     // Should never occur
-                    return H3_NULL;  // LCOV_EXCL_LINE
+                    return E_FAILED;  // LCOV_EXCL_LINE
```

Comment:
I suspect that the coverage drops are because this is the only failure case for this function, and we've added a few `if (neighborResult != E_SUCCESS)` branches after calls to this function.

It looks like we might be able to force failure here with an invalid index, with an `INVALID_DIGIT` setting as the leading non-zero digit?